app/services/analytics_events.rb

```diff
@@ -84,6 +84,13 @@ def account_visit
     track_event('Account Page Visited')
   end
 
+  # @identity.idp.event_name Add Email: Email Confirmation
+  # @param [Integer] user_id of the email confirmation
+  # A user has been sent an email confirmation
+  def add_email_confirmation(user_id, **extra)
```

Comment:
three things
1. `user_id` is going to be a string. it comes from this, where `uuid` is a string column: https://github.com/18F/identity-idp/blob/622053e2917d58170cbd4f531bbd005fca8f561d/app/services/email_confirmation_token_validator.rb#L48-L52
2. This is using a positional argument, not a keyword argument, so the callsite of this is:
    `add_email_confirmation(result.to_h)`
    which means that these values are going to be
     `user_id = { user_id: 'abcdef' }, extra = {}`
3. The comment: If I'm reading this correctly, this is called the `#create`, which is the URL we link to from inside the email template `git grep add_email_confirmation`, so that means that the user has clicked the link in a confirmation email, not that they have only been sent the link

So I think this should be:
```suggestion
  # @param [String] user_id account the email is linked to
  # A user has clicked the confirmation link in an email
  def add_email_confirmation(user_id:, **extra)
```

Reply:
I think this makes sense. I was getting a bit confused with the user_id and extra, but I see what you mean here.

Reply:
It looks like tests didn't like it. Thoughts?